app/prepare_proposal.go

```diff
@@ -32,7 +32,23 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		ante.DefaultSigVerificationGasConsumer,
 		app.IBCKeeper,
 	)
-	txs := FilterTxs(sdkCtx, handler, app.txConfig, req.BlockData.Txs)
+
+	var txs [][]byte
+	if app.LastBlockHeight() == 0 {
+		// the first block is expected to be empty; hence return an empty set of txs
+		txs = make([][]byte, 0)
+		if len(req.BlockData.Txs) != 0 {
+			// if the consensus layer sends non-empty set of transactions for
+			// block height 1, log it
+			app.Logger().Info(
```

Comment:
Why do we want to log this? i.e. what value does it bring?

Reply:
But we can conclude that it could be the case right? I'm just trying to work out what value it provides for node operators

Reply:
> But we can conclude that it could be the case right?

We expect that it would not be the case at all unless we observe this logged info. 

> I'm just trying to work out what value it provides for node operators

I understand your point. The current log doesn't tell node operators much unless they're interested in the problem of transactions in the mempool at height 0. However, having this log could help us check if this happens during local tests. I don't have a strong opinion on this though. If you think it's better, we can remove the log. Just let me know.

Reply:
Created this PR to address your comment https://github.com/celestiaorg/celestia-app/pull/2285